### PR TITLE
:seedling: Refactor e2e ownerRef test utils

### PR DESCRIPTION
**Pull request by killianmuldoon**
Update some utils in the ownerRef e2e test package to make these functions easier to import.

/area e2e-testing

### CODE REVIEW

File: test/framework/ownerreference_helpers.go [hunk 1]
```diff
@@ -168,7 +169,7 @@ var CoreOwnerReferenceAssertion = map[string]func([]metav1.OwnerReference) error
 	},
 	machineKind: func(owners []metav1.OwnerReference) error {
 		// Machines must be owned and controlled by a MachineSet or a KubeadmControlPlane, depending on if this Machine is part of a ControlPlane or not.
-		return HasOneOfExactOwners(owners, []simpleOwnerRef{machineSetController}, []simpleOwnerRef{kubeadmControlPlaneController})
```
Comment (killianmuldoon) [inline test/framework/ownerreference_helpers.go:171]: This param which is unexported was the main problem, but I think the refactor to just using ordinary ownerRefs with a custom string method makes this simpler.

Comment (sbueringer) [reply test/framework/ownerreference_helpers.go:171]: Sounds reasonable

**Pull request merged**